rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/component/ComponentController.java

```diff
@@ -132,6 +141,40 @@
         return ResponseEntity.created(location).body(halResource);
     }
 
+    @RequestMapping(value = COMPONENTS_URL + "/{componentId}/attachments", method = RequestMethod.POST, consumes = {"multipart/mixed", "multipart/form-data"})
+    public ResponseEntity<HalResource> addAttachmentToComponent(@PathVariable("componentId") String componentId, OAuth2Authentication oAuth2Authentication,
+                                                                @RequestPart("file") MultipartFile file,
+                                                                @RequestPart("attachment") Attachment newAttachment) throws TException {
+        final User sw360User = restControllerHelper.getSw360UserFromAuthentication(oAuth2Authentication);
+
+        Attachment attachment;
+        try {
+            attachment = attachmentService.uploadAttachment(file, newAttachment, sw360User);
+        } catch (IOException e) {
+            log.error(e.getMessage());
```

Comment:
You can describe the error in a short text in addition to the exception message.